Applications/Calendar/CalendarWidget.cpp

```diff
@@ -0,0 +1,169 @@
+#include "CalendarWidget.h"
+#include "AddEventDialog.h"
+#include <LibCore/DateTime.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Window.h>
+#include <LibGfx/Font.h>
+
+CalendarWidget::CalendarWidget()
+{
+    set_fill_with_background_color(true);
+    m_calendar = new Calendar(Calendar::get_year(), Calendar::get_month());
+
+    m_selected_date_label = add<GUI::Label>();
+    m_selected_date_label->set_relative_rect(10, 13, 100, 25);
+    m_selected_date_label->set_text_alignment(Gfx::TextAlignment::CenterLeft);
+    //TODO: Change text to a bigger size.
+    m_selected_date_label->set_font(Gfx::Font::default_bold_font());
+    m_selected_date_label->set_text(m_calendar->get_selected_date_text());
+
+    m_prev_month_button = add<GUI::Button>();
+    m_prev_month_button->set_text("<");
+    m_prev_month_button->set_font(Gfx::Font::default_bold_font());
+    m_prev_month_button->set_relative_rect(90, 5, 40, 40);
+    m_prev_month_button->on_click = [&] {
+        int m_target_month = m_calendar->get_selected_month() - 1;
+        int m_target_year = m_calendar->get_selected_year();
+
+        if (m_calendar->get_selected_month() <= 1) {
+            m_target_month = 12;
+            m_target_year--;
+        }
+        update_calendar_tiles(m_target_month, m_target_year);
+    };
+
+    m_next_month_button = add<GUI::Button>();
+    m_next_month_button->set_text(">");
+    m_next_month_button->set_font(Gfx::Font::default_bold_font());
+    m_next_month_button->set_relative_rect(131, 5, 40, 40);
+    m_next_month_button->on_click = [&] {
+        int m_target_month = m_calendar->get_selected_month() + 1;
+        int m_target_year = m_calendar->get_selected_year();
+
+        if (m_calendar->get_selected_month() >= 12) {
+            m_target_month = 1;
+            m_target_year++;
+        }
+        update_calendar_tiles(m_target_month, m_target_year);
+    };
+
+    m_prev_month_button = add<GUI::Button>();
+    m_prev_month_button->set_text("Add Event");
+    m_prev_month_button->set_relative_rect(475, 13, 100, 25);
+    m_prev_month_button->on_click = [&] {
+        AddEventDialog::show(m_calendar, window());
+    };
+
+    update_calendar_tiles(m_calendar->get_selected_month(), m_calendar->get_selected_year());
+}
+
+CalendarWidget::~CalendarWidget()
+{
+}
+
+void CalendarWidget::update_calendar_tiles(int target_month, int target_year)
+{
+    int i = 0;
+    //TODO: Add another row if the end of the month doesn't fit onto the current tile array.
+    for (int y = 0; y < 5; y++)
+        for (int x = 0; x < 7; x++) {
+            int x_offset = x * 85;
+            int y_offset = (y * 85) + 50;
+
+            int start_of_month = Core::DateTime::day_of_week(target_year, target_month, 1);
```

Comment:
This `DateTime` API is weird. Instead, we should instantiate a `DateTime` and ask what weekday it is:

```cpp
auto start_of_month = Core::DateTime::create(target_year, target_month, 1).weekday();
```